### DISCUSSION TITLE
Advice on engine.Mode=latency/throughput vs engine.MultiTier

### DISCUSSION THREAD

**Discussion by petenorth** [oracle/truffleruby#2135] [General]
What is current situation for TruffleRuby regarding the `engine.Mode=latency/throughput` and `engine.MultiTier` flags?

Will the multi tier compilation flag supercede `engine.Mode` ?

**Comment by eregon** [comment]
It's a good question, we need to compare results.
Right now, `Mode=latency` is used for `gem` and `bundle` (except `bundle exec`) automatically.

The plan is to make MultiTier default but this has not happened yet.

If we see MultiTier is as good as Mode=latency or very close, then I think we will just rely on MultiTier being the default and no longer explicitly use `Mode=latency` for gem/bundle, because MultiTier still reaches the best peak performance, while Mode=latency does not (it disables Truffle inlining and splitting).
`Mode=latency` might also no longer disable inlining once MultiTier is default, and then the only difference would be whether Splitting is enabled or not. In general performance is better with Splitting, but we need to ensure there is not too much splitting, which we are working on (and have been recently).
